File: backend/doc-core-service/src/main/java/com/docgen/core/domain/model/Project.java
package com.docgen.core.domain.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Project {
    private String id;
    private String name;
    private String description;
    private String repositoryUrl;
    private String ownerId;
    private List<Document> documents;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: frontend/client/src/domain/models/Project.ts
export interface Project {
  id: string;
  name: string;
  description: string;
  repositoryUrl?: string;
  ownerId: string;
  createdAt: string;
  updatedAt: string;
}

export interface CreateProjectInput {
  name: string;
  description: string;
  repositoryUrl?: string;
}
